helpers/installcredprovider.ps1

```diff
@@ -80,59 +89,65 @@ if (!$releaseId) {
 $releaseUrl = [System.IO.Path]::Combine($releaseUrlBase, $releaseId)
 $releaseUrl = $releaseUrl.Replace("\","/")
 
-$zipFile = "Microsoft.NetCore2.NuGet.CredentialProvider.zip"
+$zipFile = "Microsoft.NetCore3.NuGet.CredentialProvider.zip"
+if ($Version.StartsWith("0.")) {
+    # versions lower than 1.0.0 installed NetCore2 zip
+    $zipFile = "Microsoft.NetCore2.NuGet.CredentialProvider.zip"
+}
+if ($InstallNet6 -eq $True) {
+    $zipFile = "Microsoft.Net6.NuGet.CredentialProvider.zip"
+}
 if ($AddNetfx -eq $True) {
     $zipFile = "Microsoft.NuGet.CredentialProvider.zip"
 }
-Write-Verbose "Using $zipFile"
-
-$zipErrorString = "Unable to resolve the Credential Provider zip file from $releaseUrl"
-try {
-    Write-Host "Fetching release $releaseUrl"
-    $release = Invoke-WebRequest -UseBasicParsing $releaseUrl
-    $releaseJson = $release.Content | ConvertFrom-Json
-    $zipAsset = $releaseJson.assets | ? { $_.name -eq $zipFile }
-    $packageSourceUrl = $zipAsset.browser_download_url
-} catch {
-    Write-Error $zipErrorString
-    return
-}
 
-if (!$packageSourceUrl) {
-    Write-Error $zipErrorString
-    return
-}
+function InstallZip {
```

Comment:
I wrapped the download part from the script into a function so that it can be called twice in case the user has -AddNetfx flag and -InstallNet6 flag when calling the script. AddNetfx installs the asset that comes with the netcore3.1 version by default, so after installing the netfx we need to install the net6 one separately. Nothing inside the IstallZip function actually changes, the diff just looks a bit nasty. 

The interesting changes are at the top of the file when we decide the asset name, ($zipFile). And down below where we decide whether we need to download another asset (for net6+netfx scenario).